### PR TITLE
RFC? Extract BackgroundService abstract class from Announcement service classes

### PR DESCRIPTION
**Pull request by cpeterso**
f? @rnewman

These changes are small, but I wanted to get your feedback before submitting any bigger changes. <:) I don't have a Bugzilla bug number yet.

### CODE REVIEW

**Comment by ncalexan** [comment]
Looks good to me.

**Comment by ncalexan** [comment]
Damn adjacent buttons.

**Comment by rnewman** [comment]
Yeah, looks good to me, modulo comments above!

**Comment by cpeterso** [comment]
r? @rnewman 
f? @ncalexan 
- Removed emacs comment about 4-space tabs
- Added LOG_TAG = BackgroundService.class.getSimpleName()
- Added BackgroundService.cancelAlarm()

File: src/main/java/org/mozilla/gecko/background/BackgroundService.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    return networkInfo.isAvailable();
+  }
+
+  protected static final PendingIntent createPendingIntent(Context context, Class<? extends BroadcastReceiver> broadcastReceiverClass) {
+    final Intent service = new Intent(context, broadcastReceiverClass);
+    return PendingIntent.getBroadcast(context, 0, service, PendingIntent.FLAG_CANCEL_CURRENT);
+  }
+
+  protected final AlarmManager getAlarmManager() {
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/background/BackgroundService.java]: I would suggest that this, its sibling, and perhaps `createPendingIntent` not be `final`.

**Comment by rnewman** [comment]
r+ with `final` change. Thanks!

We merge like this:

```
# Assume you're on cpeterso/backgroundservice.
# Make sure you're up to date.
git rebase develop
# Make sure upstream is up to date, so that GitHub will auto-close pull reqs.
git push -f origin cpeterso/backgroundservice
# Merge.
git checkout develop
git merge --no-ff -m "Bug 845080 - Extract BackgroundService superclass. r=rnewman" cpeterso/backgroundservice
git push origin develop
```

Then land in `mozilla-inbound` and update bug accordingly.

File: src/main/java/org/mozilla/gecko/background/BackgroundService.java (hunk 1 above)
Comment (cpeterso) [inline src/main/java/org/mozilla/gecko/background/BackgroundService.java]: @rnewman, do you mean the `getAlarmManager()` (static and non-static) methods? Some of other methods I added are also `final`. Is the Sync coding convention to not use `final` methods or classes? I noticed the Sync code uses `final` local variables.

Comment (rnewman) [inline src/main/java/org/mozilla/gecko/background/BackgroundService.java]: We use final locals to encourage/enforce immutability (which is also polite to the compiler). Final methods prevent subclasses from overriding, which in many cases hampers testing and expected reuse. 

There are places where final classes and methods are appropriate, but a nearly abstract base class probably isn't one of them :D

**Pull request merged**